src/main/java/baritone/pathing/movement/movements/MovementPillar.java

```diff
@@ -101,6 +101,10 @@ public static double cost(CalculationContext context, int x, int y, int z) {
             // if we're standing on water and assumeWalkOnWater is false, we must have ascended to here, or sneak backplaced, so it is possible to pillar again
             return COST_INF;
         }
+        if (fromDown.getBlock() instanceof BlockLiquid && (from == Blocks.WATERLILY || from == Blocks.CARPET)) {
```

Comment:
i guess let's move the `==` to be the first check because it's unlikely (lilypad and carpet are rare in the world, water is common), and because `==` is faster than `instanceof`